vm/src/stdlib/collections.rs

```diff
@@ -426,6 +446,8 @@ mod _collections {
     #[derive(Debug)]
     struct PyDequeIterator {
         position: AtomicCell<usize>,
+        status: AtomicCell<IterStatus>,
+        length: usize, // To track length immutability.
```

Comment:
by adding 2 more fields, it looks like to be better to have `fn new(len: usize, deque: PyRef<PyDeque>)` static method.

Reply:
could also have a top level `make_deque_iter` function that takes a `reversed` flag. It might be a good refactoring-first issue for somebody on your open source contributing team.

Reply:
Merged this for now, if you can assign the refactoring to someone you're guiding that would be great, if not, tell me and fix it up.